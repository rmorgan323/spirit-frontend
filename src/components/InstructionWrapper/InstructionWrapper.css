/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.InstructionWrapper {
  display: flex;
  flex-direction: column; }
  .InstructionWrapper .patient-directions-toggle {
    margin: 5px 0;
    color: #5E378C;
    font-size: 12px;
    text-decoration: underline;
    cursor: pointer;
    z-index: 1;
    transition: all .2s ease-in-out; }
    .InstructionWrapper .patient-directions-toggle:hover {
      color: #9E87BA; }
  .InstructionWrapper .instructions-wrapper {
    max-height: 200px;
    opacity: 1;
    transition: opacity 0.3s 0.4s, max-height 0.6s ease-in-out; }
  .InstructionWrapper .hide-instructions {
    max-height: 0px;
    opacity: 0;
    transition: opacity 0.3s, max-height 0.6s 0.1s ease-in-out; }
  .InstructionWrapper .instructions {
    font-family: "Lato", sans-serif;
    font-size: 16px;
    margin: 20px auto;
    visibility: visible;
    transition: visibility 0.3s 0.4s ease-in-out; }
  .InstructionWrapper .hide-instructions .instructions {
    visibility: hidden;
    transition: visibility 0.3s ease-in-out; }
