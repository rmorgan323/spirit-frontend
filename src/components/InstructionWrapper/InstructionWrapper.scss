@import '../../variables.scss';
.InstructionWrapper {
  display: flex;
  flex-direction: column;
  .patient-directions-toggle {
    margin: 5px 0;
    color: $purple6;
    font-size: 12px;
    text-decoration: underline;
    cursor: pointer;
    z-index: 1;
    @include std-transition;
    &:hover {
      color: $purple4;
    }
  }
  .instructions-wrapper {
    max-height: 200px;
    opacity: 1;
    transition: opacity 0.3s 0.4s, max-height 0.6s ease-in-out;
  }
  .hide-instructions {
    max-height: 0px;
    opacity: 0;
    transition: opacity 0.3s, max-height 0.6s 0.1s ease-in-out;
  }
  .instructions {
    @include match-p;
    margin: 20px auto;
    visibility: visible;
    transition: visibility 0.3s 0.4s ease-in-out;
  }
  .hide-instructions .instructions {
    visibility: hidden;
    transition: visibility 0.3s ease-in-out;
  }
}