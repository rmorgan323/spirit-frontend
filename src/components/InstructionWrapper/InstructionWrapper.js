import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import './InstructionWrapper.css';

class InstructionWrapper extends Component {
  state = {
    hideInstructions: true
  };

  toggleInstructions = () =>
    this.setState(prevState => ({
      hideInstructions: !prevState.hideInstructions
    }));

  render() {
    const { children } = this.props;
    const { hideInstructions } = this.state;

    return (
      <div className="InstructionWrapper">
        <span
          className="patient-directions-toggle"
          onClick={this.toggleInstructions}
        >{`${hideInstructions ? `Show` : `Hide`} Instructions`}</span>

        <div
          className={`instructions-wrapper ${
            !hideInstructions ? `show-instructions` : `hide-instructions`
          }`}
        >
          {children}
        </div>
      </div>
    );
  }
}

export default InstructionWrapper;

InstructionWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};
