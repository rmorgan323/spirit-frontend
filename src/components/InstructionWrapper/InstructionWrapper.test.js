import React from 'react';
import InstructionWrapper from './InstructionWrapper';
import { shallow } from 'enzyme';

describe('InstructionWrapper tests', () => {
  let renderedInstructionWrapper;

  beforeEach(() => {
    renderedInstructionWrapper = shallow(
      <InstructionWrapper>
        <div>{`Instructions`}</div>
      </InstructionWrapper>
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedInstructionWrapper).toMatchSnapshot();
  });

  it('Should hide instructions by default', () => {
    expect(renderedInstructionWrapper.state('hideInstructions')).toEqual(true);
  });

  it('Should toggle hide instructions when clicked', () => {
    expect(renderedInstructionWrapper.state('hideInstructions')).toEqual(true);
    renderedInstructionWrapper.instance().toggleInstructions();
    expect(renderedInstructionWrapper.state('hideInstructions')).toEqual(false);
  });
});
