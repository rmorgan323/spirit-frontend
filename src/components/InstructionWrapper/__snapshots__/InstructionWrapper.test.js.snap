// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionWrapper tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InstructionWrapper>
    <div>
      Instructions
    </div>
  </InstructionWrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="patient-directions-toggle"
          onClick={[Function]}
        >
          Show Instructions
        </span>,
        <div
          className="instructions-wrapper hide-instructions"
        >
          <div>
            Instructions
          </div>
        </div>,
      ],
      "className": "InstructionWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show Instructions",
          "className": "patient-directions-toggle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Show Instructions",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            Instructions
          </div>,
          "className": "instructions-wrapper hide-instructions",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Instructions",
          },
          "ref": null,
          "rendered": "Instructions",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="patient-directions-toggle"
            onClick={[Function]}
          >
            Show Instructions
          </span>,
          <div
            className="instructions-wrapper hide-instructions"
          >
            <div>
              Instructions
            </div>
          </div>,
        ],
        "className": "InstructionWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show Instructions",
            "className": "patient-directions-toggle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Show Instructions",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              Instructions
            </div>,
            "className": "instructions-wrapper hide-instructions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Instructions",
            },
            "ref": null,
            "rendered": "Instructions",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
